testutil/singlevm/setup.sh

```diff
@@ -270,43 +381,96 @@ else
     echo 'dnsmasq command is not supported'
 fi
 
+# Generate the post-keystone script to issue singlevm-specific openstack
+# commands
+( cat <<-EOF
+#!/bin/bash
+
+openstack service create --name ciao compute
+openstack user create --password "$ciao_password" "$ciao_username"
+openstack role add --project service --user "$ciao_username" admin
+openstack user create --password "$ciao_demo_password" "$ciao_demo_username"
+openstack project show demo
+if [[ \$? == 1 ]]; then
+    openstack project create --domain default demo
+fi
+openstack role add --project demo --user "$ciao_demo_username" user
+openstack service create --name glance --description "Image Service" image
+openstack endpoint create --region RegionOne image public   https://$ciao_host:9292
+openstack endpoint create --region RegionOne image internal https://$ciao_host:9292
+openstack endpoint create --region RegionOne image admin    https://$ciao_host:9292
+
+# admin should only be admin of the admin project. This role was created by the
+# keystone container's bootstrap.
+openstack role remove --project service --user admin admin
```

Comment:
This takes care of admin having roles in multiple projects -- bootstrap.sh in the keystone container added admin to service, so I'll just undo that here. With this change (and #865), BAT now passes:

=== RUN   TestGetTenants
--- PASS: TestGetTenants (0.46s)
=== RUN   TestGetClusterStatus
--- PASS: TestGetClusterStatus (0.58s)
=== RUN   TestGetWorkloads
--- PASS: TestGetWorkloads (0.41s)
=== RUN   TestGetInstance
--- PASS: TestGetInstance (6.72s)
=== RUN   TestStartAllWorkloads
--- PASS: TestStartAllWorkloads (23.87s)
=== RUN   TestGetCNCIs
--- PASS: TestGetCNCIs (5.12s)
=== RUN   TestGetAllInstances
--- PASS: TestGetAllInstances (7.26s)
=== RUN   TestDeleteAllInstances
--- PASS: TestDeleteAllInstances (6.22s)
PASS
ok      github.com/01org/ciao/_release/bat/base_bat     50.840s
=== RUN   TestAddShowDelete
--- PASS: TestAddShowDelete (2.38s)
=== RUN   TestDeleteNonExisting
--- PASS: TestDeleteNonExisting (0.35s)
=== RUN   TestImageList
--- PASS: TestImageList (2.32s)
PASS
ok      github.com/01org/ciao/_release/bat/image_bat    5.052s
=== RUN   TestCreateBlockDevice
--- PASS: TestCreateBlockDevice (0.31s)
=== RUN   TestCopyBlockDevice
--- PASS: TestCopyBlockDevice (0.83s)
PASS
ok      github.com/01org/ciao/_release/bat/storage_bat  1.136s